IdentityCore/src/cache/accessor/MSIDLegacyTokenCacheAccessor.m

```diff
@@ -320,6 +354,28 @@ - (MSIDToken *)getItemForAccount:(MSIDAccount *)account
             
             return nil;
         }
+        
+        for (MSIDToken *token in tokens)
+        {
+            token.authority = authority;
```

Comment:
shouldn't we get it from the cached url?

Reply:
I think it should return the token to the developer with the authority that he requested, not the one we use internally.